source/Engine.cpp

```diff
@@ -2589,15 +2589,17 @@ void Engine::EmplaceStatusOverlays(const shared_ptr<Ship> &it,
 		(used_setting == Preferences::OverlayType::DAMAGED && !it->IsDamaged()))
 		return;
 	double width = min(it->Width(), it->Height());
+	float alpha = (used_setting == Preferences::OverlayType::ON_HIT) ? it->SmoothIsDamaged(30, 10) : 1.f;
```

Comment:
`SmoothIsDamaged` doesn't feel like something Ship needs to handle.
Just get the `TicksSinceLastHit` value with that method and figure out the appropriate alpha in here.
Alternatively, consider creating a lookup table mapping the `TicksSinceLastHit` values to alpha values, since there's not going to be that many.